core/src/test/java/com/linecorp/armeria/client/retry/RetryingHttpClientTest.java

```diff
@@ -357,6 +407,59 @@ public void shouldGetExceptionWhenFactoryIsClosed() {
                 .hasCauseInstanceOf(ClosedClientFactoryException.class);
     }
 
+    @Test
+    public void doNotRetryWhenResponseIsAborted() throws Exception {
+        final HttpClient client = client(retryAlways);
+        final HttpResponse httpResponse = client.get("/response-abort");
+        httpResponse.abort();
+
+        await().untilAsserted(() -> assertThat(responseAbortServiceCallCounter.get()).isOne());
+        // Sleep 3 seconds more to check if there was another retrying.
+        TimeUnit.SECONDS.sleep(3);
+        assertThat(responseAbortServiceCallCounter.get()).isOne();
+    }
+
+    @Test
+    public void retryDoNotStopUntilGetResponseWhenSubscriberCancel() {
+        final HttpClient client = client(retryAlways);
+        client.get("/subscriber-cancel").subscribe(
+                new Subscriber<HttpObject>() {
+                    @Override
+                    public void onSubscribe(Subscription s) {
+                        s.cancel(); // Cancel as soon as getting the subscription.
+                    }
+
+                    @Override
+                    public void onNext(HttpObject httpObject) {
+                    }
+
+                    @Override
+                    public void onError(Throwable t) {
+                    }
+
+                    @Override
+                    public void onComplete() {
+                    }
+                });
+
+        await().untilAsserted(() -> assertThat(subscriberCancelServiceCallCounter.get()).isEqualTo(3));
+    }
+
+    @Test
+    public void doNotRetryWhenRequestIsAborted() throws Exception {
+        final HttpClient client = client(retryAlways);
+
+        final HttpRequestWriter req = HttpRequest.streaming(
+                HttpHeaders.of(HttpMethod.GET, "/request-abort"));
+        req.write(HttpData.ofUtf8("I'm going to abort this request"));
+        req.abort();
+        client.execute(req);
+
+        TimeUnit.SECONDS.sleep(1);
+        // No request is made.
+        assertThat(responseAbortServiceCallCounter.get()).isZero();
```

Comment:
You don't need to call `get()` for assertion. assertj knows how to deal with `Atomic*`

Reply:
All fixed! Thank you!